### PR TITLE
[Feature] lake pk table support dup publish, multi cluster and add memory tracker

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14853

## Problem Summary(Required) ：

1. Add data version to lake primary index, and do some version check before handle txn logs. By this, we can support multi cluster and also handle same publish twice.
2. Add memory tracker to lake update manager to monitor the memory states of lake pk table.
3. Using meta cache to cache delvec.
4. Using Map in proto to record from segment id to delvec.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/del_vector.cpp [hunk 1]
```diff
@@ -120,4 +120,12 @@ void DelVector::_update_stats() {
     }
 }
 
+void DelVector::copy_from(const DelVector& delvec) {
```
Comment (sduzh) [inline be/src/storage/del_vector.cpp:123]: Why delete the copy assignment operator and add this method? Is any problem to make a custom copy constructor and copy assignment?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/del_vector.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/storage/del_vector.cpp:123]: I think add `copy_from` function is more clear that `_roaring` have been copy, not move, so I add it.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by luohaha** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/lake/update_manager.cpp [hunk 2]
```diff
@@ -401,6 +398,55 @@ void UpdateManager::remove_primary_index_cache(uint32_t tablet_id) {
...
+    if (index_entry == nullptr) {
+        // if primary index not in cache, check tablet meta directly
+        if (fs::path_exist(tablet.metadata_location(new_version))) {
+            return false;
+        } else {
+            return true;
+        }
+    } else {
+        auto& index = index_entry->value();
+        CheckVersionResult res = index.check_data_version(base_version);
```
Comment (sduzh) [inline be/src/storage/lake/update_manager.cpp:412]: IMO, it's more clear to compare the base version and index version directly:
```
int64_t index_version = index.version();
if (base_version == index_version) {
 ...
} else if (base_version < index_version) {
 ...
} else {
 ...
}
```

File: be/src/storage/lake/update_manager.cpp [hunk 3]
```diff
@@ -401,6 +398,55 @@ void UpdateManager::remove_primary_index_cache(uint32_t tablet_id) {
...
+        if (res == CheckVersionResult::SUCCESS) {
+            // success
+            _index_cache.release(index_entry);
+            return true;
+        } else if (res == CheckVersionResult::ADVANCE) {
+            // advance, remove index and return success
+            _index_cache.remove(index_entry);
```
Comment (sduzh) [inline be/src/storage/lake/update_manager.cpp:419]: Let `_index_cache.remove()` return true on success and check the return value to ensure the index really been removed.

**Comment by luohaha** [comment]
run starrocks_fe_unittest

**Comment by luohaha** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by sduzh — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**